packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -112,37 +129,11 @@ export class VaDate {
     const [year, month, day] = (this.value || '')
       .split('-')
       .map(val => parseInt(val));
-    this.setValue(year, month, day);
-
-    const daysForSelectedMonth = month > 0 ? days[month] : [];
-    const leapYear = checkLeapYear(year);
-
-    if (this.required) {
-      this.invalidYear = !year || year < minYear || year > maxYear;
-
-      this.invalidMonth = !month || month < minMonths || month > maxMonths;
 
-      this.invalidDay =
-        !this.monthYearOnly &&
-        (!day || day < minMonths || day > daysForSelectedMonth.length);
-
-      if (!leapYear && month === 2 && day > 28) {
-        this.invalidYear = true;
-        this.invalidMonth = true;
-        this.invalidDay = true;
-      }
-    }
-
-    if (
-      !this.error &&
-      (this.invalidYear || this.invalidMonth || this.invalidDay)
-    ) {
-      this.error = 'Please enter a valid date';
-    } else if (this.error !== 'Please enter a valid date') {
-      this.error;
-    } else {
-      this.error = '';
-    }
```

Comment:
This was changed and moved into a `validate()` function so that we can keep `<va-date>` and `<va-memorable-date>` as close as possible.

Reply:
Perfect, thank you.